mkosi/__init__.py

```diff
@@ -8031,25 +8031,25 @@ def generate_secure_boot_key(config: MkosiConfig) -> NoReturn:
         )
     )
 
-    cmd: List[str] = [
+    cmd: List[PathString] = [
         "openssl",
         "req",
         "-new",
         "-x509",
         "-newkey",
         f"rsa:{keylength}",
         "-keyout",
-        os.fspath(config.secure_boot_key),
+        config.secure_boot_key,
         "-out",
-        os.fspath(config.secure_boot_certificate),
+        config.secure_boot_certificate,
         "-days",
         str(config.secure_boot_valid_days),
         "-subj",
         f"/CN={cn}/",
         "-nodes",
     ]
-
-    os.execvp(cmd[0], cmd)
+    proc = run(cmd)
+    sys.exit(proc.returncode)
```

Comment:
Can we drop this? run() will throw an exception if the command fails anyway

Reply:
OK, but then we should handle that in run_verb(), and not in this function. `sys.exit()` is just bad because it skips all cleanup that may still need to happen

Reply:
`sys.exit` only raises `SystemExit`. It's not an abort like `os._exit`. Looking for example at the documentation of [atexit](https://docs.python.org/3/library/atexit.html) those are handled fine by a regular exit, i.e. `sys.exit`. Also the documentation to [`SystemExit`](https://docs.python.org/3/library/exceptions.html#SystemExit) says
> A call to [sys.exit()](https://docs.python.org/3/library/sys.html#sys.exit) is translated into an exception so that clean-up handlers ([finally](https://docs.python.org/3/reference/compound_stmts.html#finally) clauses of [try](https://docs.python.org/3/reference/compound_stmts.html#try) statements) can be executed, and so that a debugger can execute a script without running the risk of losing control.

Reply:
I still don't think a nested function should make the decision to exit the program. Let's say we want to be able to generate keys and build an image at the same time, we'd have to remove this code again.

We can just do `return generate_secure_boot_key()` in `run_verb()` and it'll work exactly the same without encoding the exit information in `generate_secure_boot_key()`